controller/routes.go

```diff
@@ -5,11 +5,32 @@ import (
 
 	"github.com/flynn/flynn/Godeps/_workspace/src/golang.org/x/net/context"
 	"github.com/flynn/flynn/controller/schema"
+	ct "github.com/flynn/flynn/controller/types"
 	"github.com/flynn/flynn/pkg/httphelper"
+	"github.com/flynn/flynn/pkg/postgres"
 	routerc "github.com/flynn/flynn/router/client"
 	"github.com/flynn/flynn/router/types"
 )
 
+func createRoute(db *postgres.DB, rc routerc.Client, appID string, route *router.Route) error {
+	route.ParentRef = routeParentRef(appID)
+	if err := schema.Validate(route); err != nil {
+		return err
+	}
+	if err := rc.CreateRoute(route); err != nil {
+		return err
+	}
+	tx, err := db.Begin()
+	if err != nil {
+		return err
+	}
+	if err := createEvent(tx.Exec, appID, route.ID, ct.EventTypeRoute, route); err != nil {
```

Comment:
`tx.Exec` returns `(sql.Result, error)` and `db.Exec` just returns `error`, the transaction allows `createEvent` to be simpler. I suppose `db.Exec` could be wrapped in a function matching the `tx.Exec` signature.